### PR TITLE
Split eggQuat

### PR DESCRIPTION
**Pull request by terorie**

### CODE REVIEW

File: sources.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 MSL_LIBC_OPTS = '-ipa file'
 RVL_OPTS = '-ipa file'
 SPY_OPTS = RVL_OPTS + " -w nounusedexpr -w nounusedarg"
-EGG_OPTS = '-ipa function -rostr'
+EGG_OPTS = '-ipa function -rostr -fp hard'
```
Comment (terorie) [inline sources.py:13]: accident

**Comment by terorie** [comment]
CI failing for unrelated reasons, build is fine.

File: source/egg/math/eggQuat.hpp [hunk 2]
```diff
@@ -7,11 +7,24 @@
...
+  void setRPY(float r, float p, float y);
+  void setAxisRotation(float angle, Vector3f* axis);
+  double sqNorm();
+  void normalise();
+  void inverse(Quatf* other);
+  void rotateVector(Vector3f* v, Vector3f* w);
+  void rotateVectorInv(Vector3f* v, Vector3f* w);
+  void mul_Vec3f(Quatf* q, Vector3f* v, Quatf* d);
+  void makeVectorRotation(Vector3f* v0, Vector3f* v1);
+  void slerpTo(Quatf* r4, Quatf* r5, double t);
```
Comment (riidefi) [inline source/egg/math/eggQuat.hpp:27]: Signature for these are `F`: `R` [`Q2` [`3EGG` `8Vector3f`]] `R` [`Q2` [`3EGG` `8Vector3f`]] and such.
`R` -> "Reference" e.g. `Vector3f&`
`P` -> "Pointer" e.g. `Vector3f*`
For code without symbols, the main difference between a ref and a ptr is that a reference cannot be null. The inputs not being nullptr is an implicit precondition of these functions: a reference makes that precondition explicit.

Comment (terorie) [reply source/egg/math/eggQuat.hpp]: fixed

**Pull request merged**